src/app/ReadClient.h

```diff
@@ -130,14 +132,30 @@ class ReadClient : public Messaging::ExchangeDelegate
         virtual void OnSubscriptionEstablished(SubscriptionId aSubscriptionId) {}
 
         /**
-         * OnResubscriptionAttempt will be called when a re-subscription has been scheduled as a result of the termination of an
-         * in-progress or previously active subscription. This object MUST continue to exist after this call is completed. The
+         * OnResubscriptionNeeded will be called when a subscription that was started with SendAutoResubscribeRequest has terminated
+         * and re-subscription is needed. The termination cause is provided to help inform subsequent re-subscription logic.
+         *
+         * The base implementation automatically re-subscribes at appropriate intervals taking the termination cause into account
+         * (see ReadClient::DefaultResubscribePolicy for more details). If the default implementation doesn't suffice, the logic of
+         * ReadClient::DefaultResubscribePolicy is broken down into its constituent methods that are publicly available for
+         * applications to call and sequence.
+         *
+         * If the method is over-ridden, it's the application's responsibility to take the appropriate steps needed to eventually
+         * call-back into the ReadClient object to schedule a re-subscription (by invoking ReadClient::ScheduleResubscription).
```

Comment:
OK, but what is the answer to my question about "consumers that want to set up CASE themselves"?  What do they need to do in practice, and how does what they need to do interact with Close/~ReadClient?

Reply:
> OK, but what is the answer to my question about "consumers that want to set up CASE themselves"?

Exactly as you mentioned above. I'll document that here.

Reply:
>  and how does what they need to do interact with Close/~ReadClient?

If they're managing the timer themselves, there is no interaction, because what I was alluding to above was the need to cancel the CASE session attempt, and to clean-up timers in `Close` or `~ReadHandler` that won't apply here.